### PR TITLE
gitops with kruise

### PR DESCRIPTION
**Pull request by zmberg**
Signed-off-by: liheng.zms <liheng.zms@alibaba-inc.com>

Document: How to publish Kruise workload (CloneSet) using GitOps (Argo-cd)?

### CODE REVIEW

File: docs/best-practices/gitops-with-kruise.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    -- check cloneSet status
+    if obj.status ~= nil then
+      if obj.status.observedGeneration < obj.metadata.generation then
+        hs.status = "Progressing"
+        hs.message = "Waiting for rollout to finish: observed cloneSet generation less then desired generation"
+        return hs
+      end
+
+      if obj.status.updatedReplicas < obj.spec.replicas then
```
Comment (furykerry) [inline docs/best-practices/gitops-with-kruise.md:151]: if updatedreadyreplicas+partition = replicas ，status should also be "Suspended"

File: docs/best-practices/gitops-with-kruise.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    ## deploy in default namespace
+    namespace: default
+```
+
+2. Create the Application CRD resource into k8s cluster, and complete the deployment using the argocd command line tool,
+refer to the [documentation](https://argo-cd.readthedocs.io/en/stable/getting_started/), as follows:
+
+![guestbook](/img/docs/best-practices/argo_sync_unknown.png)
+
+**Note: After publishing through argo-cd, we found a very strange problem. argo-cd actually did not determine whether the cloneSet publishing is complete and successful,
```
Comment (furykerry) [inline docs/best-practices/gitops-with-kruise.md:114]: in best practice document, we should be describe the solution, not the process we found the solution. So we should delete narrative sentence such as  "we found a very strange problem"

**Review by FillZpp — COMMENTED** [review]
The docs should also copied into `versioned_docs/version-v1.0/best-practices`, `i18n/zh/docusaurus-plugin-content-docs/version-v1.0/best-practices` and their sidebar file `versioned_sidebars/version-v1.0-sidebars.json`, to let users see these docs without switch version to next on the website.

File: docs/best-practices/gitops-with-kruise.md (hunk 1 above)
Comment (zmberg) [reply docs/best-practices/gitops-with-kruise.md:151]: It is estimated to be difficult to configure the partition using gitops, so the partition is not considered at this stage

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**